### PR TITLE
chore(E2E): extend `topic` fixture to allow producing messages with optional compression config

### LINKED ISSUE #2927: E2E: set up helper function for producing to a topic using compression
We can use the Javascript client to set up a producer with the `compression.codec` config set to `gzip`, `zstd`, `lz4`, and `snappy`, then include those as dimensions for the [list topics + message viewer tests](https://github.com/confluentinc/vscode/blob/main/tests/e2e/specs/topicMessageViewer.spec.ts).

See https://docs.confluent.io/kafka-clients/javascript/current/overview.html#javascript-cb-produce

### PR DESCRIPTION
**Pull request by shouples**
## Summary of Changes

While the extension doesn't offer specific functionality for producing compressed messages to a topic, users may still want to view existing compressed messages (which the extension and sidecar do support), but we don't have any way of testing that user experience extension-side. This PR adds an extra compression-specific dimension to the message viewer tests -- (`none`, `lz4`, `gzip`, `zstd`, and `snappy` based on available `compression.[codec|type]` [configs](https://docs.confluent.io/platform/current/clients/librdkafka/html/md_CONFIGURATION.html) -- along with an additional configuration in the `topicConfig` fixture option to specify producer configs, along with a utility function for producing to a target cluster's topic with some basic [options](https://github.com/confluentinc/vscode/blob/f9446719ed2e3778ae48e138de6c9b0ae3371490/tests/e2e/types/topic.ts#L24) using our Javascript [client](https://docs.confluent.io/kafka-clients/javascript/current/overview.html).

image

Closes #2927.

> [!NOTE]
> None of the assertions for the message viewer test(s) are changing in this PR, but that can be a short follow-up once we discuss as a team. 

### Optional: Any additional details or context that should be provided?

- `connectionTypes.ts` moved to `types/connection.ts` (alongside the new `types/topic.ts`) so the count of files changed is high due to adjusted imports 
- new `E2E_KAFKA_CLUSTER_NAME` environment variable used for specifying a cluster name/label when using the `topic` fixture (needed to match against any produce-message related credentials)
- new `E2E_DEBUG_LOGGING` environment variable used for including debug logs (electron/vsix setup, create-topic configs, produce-message success, etc), which cleans up the default test output
    image

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

#### Tests

- [x] Added new
- [x] Updated existing
- [ ] Deleted existing

#### Release notes

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md)?

### CODE REVIEW

File: tests/e2e/specs/schemas.spec.ts [hunk 1]
```diff
@@ -93,8 +93,8 @@ test.describe("Schema Management", { tag: [Tag.EvolveSchema] }, () => {
             });
             await expect(successNotifications.first()).toBeVisible();
 
-            const subjectLocator: Locator = schemasView.subjects.filter({ hasText: subjectName });
-            await expect(subjectLocator).toBeVisible();
+            const subjectItem: SubjectItem = await schemasView.getSubjectItem(subjectName);
+            await expect(subjectItem.locator).toBeVisible();
```
Comment (shouples) [inline tests/e2e/specs/schemas.spec.ts]: We hit a tipping point with some environments where there were too many subjects listed, so we needed to integrate the view search functionality here (leftover from https://github.com/confluentinc/vscode/pull/3138)

File: tests/e2e/utils/producer.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    if (
+      directConnectionConfig?.kafkaConfig &&
+      directConnectionConfig.kafkaConfig.authType !== SupportedAuthType.None
+    ) {
+      const credentials = directConnectionConfig.kafkaConfig.credentials;
+      saslConfig = {
+        mechanism: "plain",
+        username: credentials.api_key || credentials.username,
+        password: credentials.api_secret || credentials.password,
+      };
```
[the remark this answers is not in the document]
Comment (shouples) [reply tests/e2e/utils/producer.ts]: The CCloud connection will inherit the default direct connection [configs](https://github.com/confluentinc/vscode/blob/1f0684c004b5593d080843965e94ff1384c77f51/tests/e2e/baseTest.ts#L186-L205), so this is fine

File: tests/e2e/objects/views/View.ts [hunk 3]
```diff
@@ -115,29 +130,13 @@ export class SearchableView extends View {
...
-      return itemLocator;
-    } catch {
-      // item not found, try searching for it
-    }
-
-    await this.search(label);
-
     await expect(itemLocator).toBeVisible();
     return itemLocator;
   }
```
[the remark this answers is not in the document]
Comment (shouples) [reply tests/e2e/objects/views/View.ts]: The search behavior is fast enough since the items are all cached after initial load, and the JSDoc comment for the method was updated

File: tests/e2e/specs/topicMessageViewer.spec.ts [hunk 4]
```diff
@@ -32,34 +33,54 @@ test.describe("Topics Listing & Message Viewer", { tag: [Tag.TopicMessageViewer]
     SelectKafkaCluster.FromResourcesView,
     SelectKafkaCluster.FromTopicsViewButton,
   ];
+  const compressionTypes: KafkaJS.CompressionTypes[] = [
+    KafkaJS.CompressionTypes.None,
+    KafkaJS.CompressionTypes.GZIP,
+    KafkaJS.CompressionTypes.Snappy,
+    KafkaJS.CompressionTypes.LZ4,
+    KafkaJS.CompressionTypes.ZSTD,
+  ];
```
[the remark this answers is not in the document]
Comment (shouples) [reply tests/e2e/specs/topicMessageViewer.spec.ts]: Oh we're testing all of it.

File: tests/e2e/objects/views/View.ts [hunk 5]
```diff
@@ -61,7 +61,10 @@ export class View {
     // when the header is collapsed (even though the header is always visible by default)
     await this.ensureExpanded();
     await this.header.hover();
-    await this.header.getByRole("button", { name: label }).click();
+    const action = this.header.getByRole("button", { name: label });
+    // avoid race condition where action isn't visible before clicking
+    await expect(action).toBeVisible();
+    await action.click();
```
[the remark this answers is not in the document]
Comment (shouples) [reply tests/e2e/objects/views/View.ts]: https://github.com/confluentinc/vscode/pull/3021#discussion_r2691994789

**Review by Cerchie — APPROVED** [review]
